homeassistant/helpers/update_coordinator.py

```diff
@@ -465,13 +467,23 @@ def _async_refresh_finished(self) -> None:
 class BaseCoordinatorEntity(entity.Entity, Generic[_BaseDataUpdateCoordinatorT]):
     """Base class for all Coordinator entities."""
 
+    _no_use_coordinator = False
+
     def __init__(
         self, coordinator: _BaseDataUpdateCoordinatorT, context: Any = None
     ) -> None:
         """Create the entity with a DataUpdateCoordinator."""
         self.coordinator = coordinator
         self.coordinator_context = context
 
+    def __init_subclass__(cls, **kwargs: Any) -> None:
+        """Post initialisation processing."""
+        super().__init_subclass__(**kwargs)
+        # ButtonEntity and EventEntity provide their own states
+        # and should not register to use the coordinator for state updates.
+        if ButtonEntity in cls.__mro__ or EventEntity in cls.__mro__:
```

Comment:
I don't think the coordinator should know about platforms. It's up to each integration to implement the coordinator entity or not do that.

Reply:
Comes from the side-discussion [here](https://discord.com/channels/330944238910963714/1222192972519706798/1233166977292308561) in discord.
It just creates unnecessary overhead by registering listeners for entities which has no use of it.

Reply:
Integrations shouldn't use the coordinator entity if they shouldn't use it.